### PR TITLE
[MM-19422] Add LDAP trace output to console and server logs

### PR DESCRIPTION
**Pull request by hahmadia**
Now you get logging printed to mLog
Example:
```
{"level":"error","ts":1574371628.405395,"caller":"log/log.go:172","msg":"LDAP Request: (Universal, Constructed, Sequence and Sequence of) Len=170 \"<nil>\"\n MessageID: (Universal, Primitive, Integer) Len=1 \"2\"\n Search Request: (Application, Constructed, 0x03) Len=164 \"<nil>\"\n  Base DN: (Universal, Primitive, Octet String) Len=20 \"dc=mm,dc=test,dc=com\"\n  Scope: (Universal, Primitive, Enumerated) Len=1 \"2\"\n  Deref Aliases: (Universal, Primitive, Enumerated) Len=1 \"3\"\n  Size Limit: (Universal, Primitive, Integer) Len=1 \"0\"\n  Time Limit: (Universal, Primitive, Integer) Len=1 \"60\"\n  Types Only: (Universal, Primitive, Boolean) Len=1 \"false\"\n  Or: (Context, Constructed, 0x01) Len=86 \"<nil>\"\n   Equality Match: (Context, Constructed, 0x03) Len=20 \"<nil>\"\n    Attribute: (Universal, Primitive, Octet String) Len=11 \"objectClass\"\n    Condition: (Universal, Primitive, Octet String) Len=5 \"group\"\n   Equality Match: (Context, Constructed, 0x03) Len=27 \"<nil>\"\n    Attribute: (Universal, Primitive, Octet String) Len=11 \"objectClass\"\n    Condition: (Universal, Primitive, Octet String) Len=12 \"groupOfNames\"\n   Equality Match: (Context, Constructed, 0x03) Len=33 \"<nil>\"\n    Attribute: (Universal, Primitive, Octet String) Len=11 \"objectClass\"\n    Condition: (Universal, Primitive, Octet String) Len=18 \"groupOfUniqueNames\"\n  Attributes: (Universal, Constructed, Sequence and Sequence of) Len=37 \"<nil>\"\n   Attribute: (Universal, Primitive, Octet String) Len=9 \"entryUUID\"\n   Attribute: (Universal, Primitive, Octet String) Len=2 \"cn\"\n   Attribute: (Universal, Primitive, Octet String) Len=6 \"member\"\n   Attribute: (Universal, Primitive, Octet String) Len=12 \"uniqueMember\"","source":"stdlog"}
```

### CODE REVIEW

**Comment by hanzei** [comment]
@hahmadia Is this ready for review?

**Comment by hahmadia** [comment]
> @hahmadia Is this ready for review?

Not yet. I couldn't find a WIP label but I will add the reviewers after tomorrow hackathon presentation. Sorry just been busy with hackathon so didn't have time to update this PR to mention it's in progress.

**Comment by hanzei** [comment]
No problem. I just created and added a WIP label.

**Review by sbishel — APPROVED** [review]

File: debug.go [hunk 1]
```diff
@@ -18,13 +19,17 @@ func (debug *debugging) Enable(b bool) {
...
 	}
 }
 
 // PrintPacket dumps a packet.
 func (debug debugging) PrintPacket(packet *ber.Packet) {
 	if debug {
-		ber.PrintPacket(packet)
+		var b bytes.Buffer
+		ber.WritePacket(&b, packet)
+		textToPrint := "ldap-trace: " + b.String()
```
Comment (mkraft) [inline debug.go]: Could we set `"ldap-trace: "` as a constant once?

**Review by mkraft — CHANGES_REQUESTED** [review]

File: debug.go (hunk 1 above)
Comment (hahmadia) [reply debug.go]: Thanks for suggestion! Done.

**Review by mkraft — APPROVED** [review]

**Comment by mkraft** [comment]
Oops. Should this have had a 'QA Review' and not been merged?

**Pull request merged**